includes/orders/classes/class-order.php

```diff
@@ -355,6 +355,39 @@ public function get_items() {
 		return $this->items;
 	}
 
+
+	/**
+	 * Retrieve all the items in the order and transform bundle items into regular items.
+	 *
+	 * @since 3.0.2
+	 *
+	 * @return Order_Item[] Order items.
+	 */
+	public function get_items_with_bundles() {
```

Comment:
I don't love the name here. I feel like we could lock ourselves into it just being used for `bundles`.

Let's try and not limit ourselves in the future. Maybe name it something like `get_deliverable_items`.

This would give us the freedom (by name only, but make it more clear) that we are returning all products that this order gives access to, leaving room for filters later in the future.

Reply:
My concern with get_deliverable_items is that we have a similarly named helper method which returns specific order item statuses. Using this name implies that we are returning items which have a deliverable status, I think.

I'm wondering if we could add a parameter to the get_items method and add this functionality in there instead? If we allowed an array as the parameter, then we could (for today) set a flag to include bundled items. Eventually we could include a status flag, to return only deliverable items. Not sure this is the right way to go but throwing out options.

Reply:
@robincornett I wondered about that too but didn't want to drastically change that function immediately as those are actual items purchased, this is a different context than just getting the actual order items.

Let's just stick with this now and we can maybe work on adjusting at a later date.